pypicloud/access/ldap_.py

```diff
@@ -141,10 +152,15 @@ def verify_user(self, username, password):
         else:
             return True
         finally:
-            self._server.simple_bind_s(
-                self._service_dn,
-                self._service_password
-            )
+            if self._service_dn:
+                # bind with the service_dn
+                self._server.simple_bind_s(
+                    self._service_dn,
+                    self._service_password
+                )
+            else:
+                # force a connection without binding
+                self._server.whoami_s()
```

Comment:
Since this "if...else..." logic is used twice, can you factor it out into a method and call it in both places?